### ISSUE TITLE
MSGSRV-A = MSGSRV-J

### ISSUE THREAD

**Issue by MatthiasL-SAP** [SAP-samples/frun-csa-policies-best-practices#16]
When you look at chapter 2.2.1.4.2 "[MSGSRV-J]: Message Server Security" it says:  The rules from MSGSRV-A apply for Java systems as well.

However when we look at [MSGSRV-A](https://github.com/SAP-samples/frun-csa-policies-best-practices/blob/main/BaselinePolicies/SOS/v2.2/ABAP_SYSTEM/BL2_MSGSRV-A.xml) it seems to be restricted only for ABAP_INSTANCE_PAHI, at least it isn't giving any hits for a JAVA-only system.  

I can however find them when I use the "Find in Configuration Data" and find them in the config store S00165 (and/or S00439). Perhaps it's an idea to make them allow for both?

**Comment by MatthiasL-SAP** [comment]
Additionally, the same applies to RFCGW_A = RFCGW_J

**Comment by FrankBuchholz** [comment]
You find the profile parameters in configuration store `ABAP_INSTANCE_PAHI` for ABAP based systems and in configuration store `Parameters` for Java based systems.

The policy [BL2_MSGSRV-A](https://github.com/SAP-samples/frun-csa-policies-best-practices/blob/main/BaselinePolicies/SOS/v2.2/ABAP_SYSTEM/BL2_MSGSRV-A.xml) contains the rules for ABAP based systems whereas the policy [BL2_SYSTEM-J](https://github.com/SAP-samples/frun-csa-policies-best-practices/blob/main/BaselinePolicies/SOS/v2.2/JAVA_ALL/BL2_SYSTEM-J.xml) contains the rules for Java based systems.

Example:

ABAP
```
  <configstore name="ABAP_INSTANCE_PAHI" sci_id="">
    <checkitem desc="[p2-STANDARD] Defined internal port for server communication" id="MSGSRV-A_a">
      <compliant>NAME = 'rdisp/msserv_internal'  and lpad(VALUE, 4, '0') &gt; '0000' </compliant>
      <complianttext/>
      <noncompliant>NAME = 'rdisp/msserv_internal'  and not ( lpad(VALUE, 4, '0') &gt; '0000' ) </noncompliant>
      <noncomplianttext/>
    </checkitem>
```

Java
```
  <configstore name="Parameters" sci_id="S00165">
    <!-- MSGSRV-J        Message Server Security - AS JAVA -->
    <checkitem desc="[p2-STANDARD] Defined internal port for server communication" id="MSGSRV-J_a">
      <compliant>    NAME = 'rdisp/msserv_internal'  and VALUE not like_regexpr '^\s*$' </compliant>
      <complianttext/>
      <noncompliant> NAME = 'rdisp/msserv_internal'  and VALUE like_regexpr '^\s*$' </noncompliant>
      <noncomplianttext/>
    </checkitem>
```

(The check in this example is not identical but produces the same result.)

**Comment by MatthiasL-SAP** [comment]
Ah I see now, I was searching for RFCGW-J but they weren't renamed in the JAVA-version. So unlike in the above examle, I need to look for 'RFCGW-A' in [BL2_APPLICATION-J.xml](https://github.com/SAP-samples/frun-csa-policies-best-practices/blob/main/BaselinePolicies/SOS/v2.2/JAVA_ALL/BL2_APPLICATION-J.xml).

**Comment by FrankBuchholz** [comment]
By the way:
a) We just have published version 2.5 of the Security Baseline Document at [https://support.sap.com/sos](https://support.sap.com/sos) (but there is no corresponding package for FRUN yet) in April 2024
b) The published FRUN package for v2.4 already shows some optimized policy files for AS Java at [https://github.com/SAP-samples/frun-csa-policies-best-practices/tree/main/BaselinePolicies/SOS/v2.4/JAVA_ALL](https://github.com/SAP-samples/frun-csa-policies-best-practices/tree/main/BaselinePolicies/SOS/v2.4/JAVA_ALL) (but your suggestions about self registration are not covered yet.)

**Comment by ManfredAch** [comment]
Java policies have been adjusted.
Currently the new version related to baseline 2.5 can't be provided.
